test/tools/router_check/router.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "common/common/logger.h"
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+#include "common/router/config_impl.h"
+
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/precompiled/precompiled_test.h"
+#include "test/test_common/utility.h"
+#include "test/tools/router_check/json/tool_config_schemas.h"
+
+/**
+ * Class that stores the configuration parameters of the router
+ * check tool extracted from a json input file.
+ */
+struct ToolConfig {
+  ToolConfig() : random_value_(0){};
+  void parseFromJson(const Json::ObjectPtr& check_config);
+
+  int random_value_;
+  Http::TestHeaderMapImpl headers_;
+  Router::RouteConstSharedPtr route_;
+};
+
+/**
+ * Router check tool to check routes returned by a router
+ */
+class RouterCheckTool : Logger::Loggable<Logger::Id::testing> {
+public:
+  /**
+   * @param config_json specifies the json config file to be loaded
+   * @param schema is the json schema to validate against
+   * @return Json::ObjectPtr pointer to router config json object. Return
+   * nullptr if load fails.
+   */
+  Json::ObjectPtr loadJson(const std::string& config_json, const std::string& schema);
+
+  /**
+   * @param router_config_json specifies the router config json file
+   * @return bool if json file loaded successfully and ConfigImpl object created
+   * successfully
```

Comment:
nit: spacing. Can you aline successfully under bool? like https://github.com/lyft/envoy/blob/master/include/envoy/http/async_client.h#L133-L135